### DISCUSSION TITLE
output-directory=build except for PdF? (Linux Mint)

### DISCUSSION THREAD

**Discussion by ReaderGuy42** [texstudio-org/texstudio#2824] [Q&A]
I've figured out that if I put `pdflatex -synctex=1 -interaction=nonstopmode -output-directory=build %.tex` into the PdfLaTeX Command Line, it puts almost all the files into a folder (Sidenote: is there a way to have TeXStudio make that folder? So far I've only seen this work if the folder is already there), and this in the BibTeX line: `bibtex build/% -include-directory=build`.

There is still a `.bbl` and a `.blg` file, can those somehow be generated in the `build` folder too?

I also just realized that TeXStudio is no longer finding all the auxiliary files it just created, even though I also added `./build/` to the Log Files Search Path. How do I deal with this?

The original issue was to keep the PDF file in the main directory. How do I achieve this? I've tried using `-aux-directory` but that gives me an `unknown option` error, so that might not be possible for me.

Ideally, I'd like to just have the `.tex` file and the PDF in the main directory, and everything else in the `build` folder. Is this possible? I'm on Linux Mint.

**Comment by foreachthing** [comment]
Maybe this will help: https://tex.stackexchange.com/questions/269431/tutorial-texstudio-add-output-path/269432#269432

**Comment by muzimuzhi** [comment]
> [...] and this in the BibTeX line: `bibtex build/% -include-directory=build`.

This works for me: `sh -c "cd build; bibtex %.aux"`

> I also just realized that TeXStudio is no longer finding all the auxiliary files it just created, even though I also added `./build/` to the Log Files Search Path. How do I deal with this?

Adding just `build` to the Log Files Search Path works for me. Is is possible that your Log Files field contains multiple paths and wrong path separator was used? 

> Ideally, I'd like to just have the `.tex` file and the PDF in the main directory, and everything else in the `build` folder. Is this possible? I'm on Linux Mint.

`-output-directory=dir` will put everything, including the pdf file to `dir`. You may need to move pdf file out.

MiKTeX comes with an `-aux-directory=dir`, and `latexmk` provides a proper emulation for this option in TeX Live. Try `latexmk -pdflatex -aux-directory=build main.tex`.

Currently `latexmk` also moves `fls` files to current working directory, which is inconsistent with MiKTeX v20.10 or newer, see MiKTeX/miktex#634 and MiKTeX/miktex#914. I've reported it to `latexmk`'s maintainer.

**Reply by muzimuzhi** [reply]
> Currently `latexmk` also moves `fls` files to current working directory, which is inconsistent with MiKTeX v20.10 or newer, see [MiKTeX/miktex#634](https://github.com/MiKTeX/miktex/issues/634) and [MiKTeX/miktex#914](https://github.com/MiKTeX/miktex/issues/914). I've reported it to `latexmk`'s maintainer.

Since v4.79, `latexmk` moves `fls` to aux directory by default. One can achieve previous behavior by setting new variable `$fls_uses_aux_dir` to non-zero value. See full info in `texdoc latexmk`, searching for "Location of .fls file".
